### DISCUSSION TITLE
How to keep a language server alive in background when all related files are closed

### DISCUSSION THREAD

**Discussion by harry-xm** [redhat-developer/lsp4ij#487] [Q&A] [answered]
In a large project, it might take quite some time for rust-analyzer to initialize itself.

If someone clicked "Close All Tabs" and subsequently opens another file, the language server is restarted and I have to wait until it finishes loading.

**Comment by angelozerr** [comment, marked as answer]
Indeed this behavior comes from https://github.com/eclipse/lsp4e the LSP support for Eclipse (LSP4IJ has been created from this project and adapted for IJ).

I understand that it is annoying, please create an issue to customize that with an API.
